src/mysql/MySQL.chpl

```diff
@@ -0,0 +1,677 @@
+        var _cptr_fields: c_ptr(MYSQL_FIELD);
+        var _fieldTypes: string;
+        var valid: bool;
+
+        pragma "no doc"
+        proc init(valid: bool) {
+            this._cptr_fields = nil;
+            this.valid = false;
+        }
+
+        pragma "no doc"
+        proc init(mysqlrow: MYSQL_ROW, cfields: c_ptr(MYSQL_FIELD), valid: bool = true) {
+            this._cptr_row = mysqlrow;
+            this._cptr_fields = cfields;
+
+            this.valid = valid;
+            // TODO: init _fieldTypes here
+        }
+
+        proc isValid() {
+            return this.valid;
+        }
+
+        /*
+        Returns the value at the field number (index) specified by fieldNumber as
+        the specified type. The first field is the zeroth field.
+        Note that this function and its overload is valid only for those types
+        that can be converted from string to that type.
+        To handle the conversion yourself, use `getVal()` instead, which returns a string.
+            :arg fieldNumber: the nth field whose value is required in the row (n starts from 0)
+            :type fieldNumber: int(32)
+            :arg t: The type to which the field value should be converted to while returning the value.
+            :type t: type
+            :return: the value of the fieldNumber-th field converted to type t
+            :rtype: t
+        */
+        override proc getValAsType(fieldNumber: int(32), type t) {
+            if (!this.isValid()) {
+                throw new InvalidRowError();
+            }
+            var fieldVal: string = createStringWithNewBuffer(__get_mysql_field_val_by_number(this._cptr_row, fieldNumber));
+            return fieldVal: t;
+        }
+
+        /*
+        Returns the value for the field name specified by fieldName as
+        the specified type.
+        Note that this function and its overload is valid only for those types
+        that can be converted from string to that type.
+        To handle the conversion yourself, use `getVal()` instead, which returns a string.
+            :arg fieldName: the name of the field whose value is required in the row (n starts from 0)
+            :type fieldName: string
+            :arg t: The type to which the field value should be converted to while returning the value.
+            :type t: type
+            :return: the value of the field with name fieldName converted to type t
+            :rtype: t
+        */
+        override proc getValAsType(fieldName: string, type t) {
+            if (!this.isValid()) {
+                throw new InvalidRowError();
+            }
+            var fieldVal: string = createStringWithNewBuffer(__get_mysql_field_val_by_name(this._cptr_row, 
+                                                                                           this._cptr_fields, 
+                                                                                           fieldName.localize().c_str()));
+            return fieldVal: t;
```

Comment:
What if it can't be typecasted? Shouldn't we throw an error?

Reply:
Since the `return` uses Chapel's explicit type conversion, an `IllegalArgumentError: bad cast from string <string value> to <type>` will be thrown automatically in such a case.